mlflow/spark/__init__.py

```diff
@@ -992,26 +992,66 @@ def predict(
             # Spark model uses "prediction" as default model inference output column name.
             return self.spark_model.transform(pandas_df)["prediction"]
 
+        # Convert List[np.float64] / np.array[np.float64] type to List[float] type,
+        # otherwise it will break `spark.createDataFrame` column type inferring.
+        if self.signature.inputs:
+            for col_spec in self.signature.inputs.inputs:
+                if isinstance(col_spec.type, Array) and col_spec.type.is_sparkml_vector:
+                    if col_spec.name is None:
+                        col_name = pandas_df.columns[0]
+                    else:
+                        col_name = col_spec.name
+
+                    pandas_df[col_name] = pd.Series([
+                        [float(elem) for elem in array]
+                        for array in pandas_df[col_name]
+                    ])
+
+        spark_df = self.spark.createDataFrame(pandas_df)
+
+        # Convert Array[Double] column to spark ML vector type according to signature
+        if self.signature.inputs:
+            for col_spec in self.signature.inputs.inputs:
+                if isinstance(col_spec.type, Array) and col_spec.type.is_sparkml_vector:
```

Comment:
Same here with accessing this member `is_sparkml_vector`

Reply:
https://github.com/mlflow/mlflow/pull/12758#discussion_r1689031153